docs/en/manuals/extensions_details.md

```diff
@@ -0,0 +1,79 @@
+
+# The Defold build setup
+
+Here we list some relevant build information, in order to make the integrations with your extensions as easy as possible.
+
+Here are some things to consider when you create an extension for the Defold engine.
+
+# Static linkage
+
+The custom engine is built using static linkage.
+The main reason is that on iOS, multiple executable binaries in an .ipa aren't allowed in the app store.
+
```

Comment:
Maybe add something about not polluting the global symbol namespace; keep your helper functions `static` and/or put them in an anonymous namespace?

Reply:
Yes, that are good points.

Perhaps we need more guidelines regarding "good code" and avoiding namespace clashing in general? And by good code I don't mean C vs C++ but things as @erikangelin-king mentions, if we don't already have that somewhere.